.github/workflows/ci.yaml

```diff
@@ -93,6 +93,17 @@ jobs:
         with:
           version: v1.55.2
           working-directory: ${{ matrix.directory }}
+          args: -E gosec -E gofmt --timeout=10m
```

Comment:
Is this related?

Reply:
Yes, Instead of adding gosec as a seperate action, it can be passed as an argument in golangci-lint as well. Both give the same result. Moreover from what i have read, golangci-lint has default settings to reduce the amount of false positives raised by gosec and other linters. 
If we intend to use default settings and don't require to make changes to rules of gosec, above option can be opted.
And gofmt is added as it is standard and will auto-fix the formatting errors.

Reply:
@Sheetalpamecha Thanks for explaining. So we have 3 unrelated changes in this commit:
- Adding the voln checker
- Adding gosec checker via golangci-lint
- Adding formatting support to golangci-lint

The voln checker part looks good to me, but the issue or this commit do not explain why we need it, and how it play with github built in security checks, so I cannot approve this change.

Adding gosec checker sounds like a good idea, but mixing security related checks with linting - which is mostly about keeping the code consistent according to the project style is really bad. I would like to see the gosec checker as a separate step, with its own configuration so we can avoid false positives and other issues with the tool.

Adding formatting support to golangci-lint sounds interesting, but it depends on the implementation. I would like to see examples the output created by the -E gofmt, and understand how it mixes in github CI. For example I hope that we will get an error if the code is not formatted by gofmt style, and the error would include the formatted code.

Both security related changes can be in the same PR, one commit per change, maybe using a new security check job, so we can share the setup go work. Including gofmt in golangci-lint should be in another PR since it is not related to security.

Reply:
Sure @nirs, It does make sense to separate all the 3 changes in different commits. For this PR I will update the commits with the security changes and description on their use case and will open a separate PR for gofmt.

Reply:
`gofmt` and `gosec` are enabled in the golangci yaml, and do not need to be enabled explicitly.
https://github.com/RamenDR/ramen/blob/883b9ed91b2ee5b1b47ba6f7402066f0bb180666/.golangci.yaml#L123
https://github.com/RamenDR/ramen/blob/883b9ed91b2ee5b1b47ba6f7402066f0bb180666/.golangci.yaml#L129